### PR TITLE
Feature/specdef refactor

### PR DESCRIPTION
**Pull request by ens-ap5**
## Requirements

- Filling out the template is required. Any pull request that does not include enough information to be efficiently reviewed may be rejected.
- Please consider which branch this is to be submitted against. This is usually obvious, however if it is to be applied to both a release branch nn and master then please submit it against postreleasefix/nn and let us merge it into to the two branches.

## Description

Refactor of ConfigPacker.pm, to remove duplicate code and therefore make maintenance easier in future.

## Views affected

Any views that include data compiled by ConfigPacker, but particularly the "Configure this Page" track lists and any compara alignment selectors.

## Possible complications

This is low-level code that affects not just the vertebrate site but any Ensembl site (non-vertebrates, Rapid Release, etc). Also, this code cannot be merged until the corresponding changes to eg-web-common have also been reviewed - certain pieces of code have been pulled in from the latter repo where appropriate.

## Merge conflicts

No merge conflicts - no-one else should be working on this module!

## Related JIRA Issues (EBI developers only)

https://www.ebi.ac.uk/panda/jira/browse/ENSWEB-6168

### CODE REVIEW

**Comment by ens-ma7** [comment]
hi @ens-ap5  is it possible to have a sandbox URL.

File: modules/EnsEMBL/Web/ConfigPacker.pm [hunk 1]
```diff
@@ -1361,6 +1364,33 @@ sub _summarise_compara_db {
...
+        and mls.method_link_id = ml.method_link_id
+        and ml.class = "GenomicAlignBlock.pairwise_alignment"
+      group by mls.method_link_species_set_id, mls.method_link_id
+      having count = 1
+  );
+}
+
+sub _homologies_sql {
+  return qq(
+    select ml.type, gd.name, gd.name, count(*) as count
```
Comment (ens-ma7) [inline modules/EnsEMBL/Web/ConfigPacker.pm:1383]: there is double gd.name in the select statement, i know it was just moving the code but i am not sure if we need gd.name twice, might be good to remove one of it unless it is supposed to be another column name.

Comment (ens-ap5) [reply modules/EnsEMBL/Web/ConfigPacker.pm:1383]: Yes, it's necessary. The code is combining two arrayrefs before it builds the hash - see line 1347. If we didn't have the name twice, the paralogues hash would be incomplete.

**Comment by ens-ap5** [comment]
> hi @ens-ap5 is it possible to have a sandbox URL.

http://wp-np2-1f.ebi.ac.uk:10105/index.html

**Review by jyothishnt — APPROVED** [review]
Species selector works on Compara and Multi. I guess it should work on tools too. No obvious issues I could see in the code. Approving. Thanks Anne.

**Comment by ens-ma7** [comment]
I have clicked through the sandbox URL and all look fine. I am happy with the changes. But they are definitely big changes and best to have this ready on staging for testing.

**Review by ens-ma7 — APPROVED** [review]

**Comment by ens-ap5** [comment]
Sure - I'll merge them today.

**Pull request merged**